app/src/main/java/info/nightscout/androidaps/plugins/general/overview/OverviewFragment.kt

```diff
@@ -425,6 +429,22 @@ class OverviewFragment : DaggerFragment(), View.OnClickListener, OnLongClickList
                         }.show(childFragmentManager, "Overview")
                     })
                 }
+
+                R.id.cannula_or_patch_layout -> {
+                    activity.let { protectionCheck.queryProtection(activity, ProtectionCheck.Protection.BOLUS, UIRunnable { FillDialog().show(childFragmentManager, "FillDialog") }) }
+                }
+
+                R.id.insulin_layout -> {
+                    activity.let { protectionCheck.queryProtection(activity, ProtectionCheck.Protection.BOLUS, UIRunnable { FillDialog().show(childFragmentManager, "FillDialog") }) }
```

Comment:
patch pumps do not fill, they only replace

Reply:
add conditions